packages/auto-approve/src/auto-approve.ts

```diff
@@ -0,0 +1,226 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file handles the logic to manage incoming pull-requests
+
+// eslint-disable-next-line node/no-extraneous-import
+import {Probot, Context, ProbotOctokit} from 'probot';
+import {logger} from 'gcf-utils';
+import {ValidPr, checkPRAgainstConfig} from './check-pr';
+import {getChangedFiles, getBlobFromPRFiles} from './get-PR-info';
+import {validateYaml, validateSchema, checkCodeOwners} from './check-config.js';
+
+interface Configuration {
+  rules: ValidPr[];
+}
+const CONFIGURATION_FILE_PATH = 'auto-approve.yml';
+
+/**
+ * Takes in the auto-approve.yml file and (if it exists) the CODEOWNERS file
+ * and checks both files to ensure they match schema, format, and appropriate CODEOWNERS
+ *
+ * @param owner owner of the repo of the incoming PR
+ * @param repo string, the name of the repo of the incoming PR
+ * @param codeOwnersFile string, the CODEOWNERS file in .github/CODEOWNERS for the repo of the incoming PR
+ * @param octokit the octokit instance
+ * @param headSha the sha upon which to check whether the config and the CODEOWNERS file are configured correctly
+ * @returns true if the status check passed, false otherwise
+ */
+export async function logicForConfigCheck(
+  owner: string,
+  repo: string,
+  config: string,
+  codeOwnersFile: string | undefined,
+  octokit: InstanceType<typeof ProbotOctokit>,
+  headSha: string
+): Promise<Boolean> {
+  // Check if the YAML is formatted correctly
+  const isYamlValid = validateYaml(config);
+
+  // Check if config has correct schema
+  const isSchemaValid = await validateSchema(config);
+
+  // Check if codeowners includes @github-automation for auto-approve.yml file
+  const isCodeOwnersCorrect = await checkCodeOwners(
+    octokit,
+    owner,
+    repo,
+    codeOwnersFile
+  );
+
+  // If all files are correct, then submit a passing check for the config
+  if (
+    isYamlValid === true &&
+    isSchemaValid === true &&
+    isCodeOwnersCorrect === true
+  ) {
+    await octokit.checks.create({
+      owner,
+      repo,
+      head_sha: headSha,
+      name: 'Auto-approve.yml check',
+      conclusion: 'success',
+      output: {
+        title: 'Auto-approve.yml check',
+        summary: 'Successful auto-approve.yml config check',
+        text: '',
+      },
+    });
+
+    return true;
+  } else {
+    // If any of the files are not correct, submit a failing check
+    // logging the appropriate error messages
+    const errorMessage =
+      'See the following errors in your auto-approve.yml config:\n' +
+      `${
+        typeof isCodeOwnersCorrect === 'string' ? isCodeOwnersCorrect : ''
+      }\n` +
+      `${typeof isYamlValid === 'string' ? isYamlValid : ''}\n` +
+      `${
+        typeof isSchemaValid === 'object'
+          ? JSON.stringify(isSchemaValid, null, 2)
+          : ''
+      }`;
+
+    await octokit.checks.create({
```

Comment:
Along with splitting out into specific validation methods, I'd probably make the actual adding a `check` happen outside of the validation, that way you can perform validation in isolation without a side effect.

Reply:
Same comment as above... let me know if you still think it's worth further dividing!